crates/sozo/src/commands/options/account.rs

```diff
@@ -14,12 +14,16 @@ use super::{
 
 #[derive(Debug, Args)]
 #[command(next_help_heading = "Account options")]
+// INVARIANT:
+// - For commandline: we can either specify `private_key` or `keystore_path` along with
+//   `keystore_password`. This is enforced by Clap.
+// - For `Scarb.toml`: if both private_key and keystore are specified in `Scarb.toml` private_key
+//   will take priority
 pub struct AccountOptions {
     #[arg(long, env = DOJO_ACCOUNT_ADDRESS_ENV_VAR)]
     pub account_address: Option<FieldElement>,
 
     #[arg(long, env = DOJO_PRIVATE_KEY_ENV_VAR)]
-    #[arg(requires = "account_address")]
```

Comment:
removed to be consistent with the behaviour of `keystore`.

user can specify `account_address` in `Scarb.toml` and provide `private_key` from commandline.